cpp/src/command_classes/UserCode.cpp

```diff
@@ -264,7 +264,8 @@ bool UserCode::HandleMsg
 				memcpy( data, &_data[3], size );
 				value->OnValueRefreshed( data, size );
 			} else {
-				value->OnValueRefreshed( 0x00, 1);
+				size = 0;
+				value->OnValueRefreshed( (uint8 const*)&size, 1);
 			}
```

Comment:
Thanks, Dave,

To follow your CI comment, should the procedure be:

- close the PR858
- edit the code
- create a new PR

Regards
Bin

On 16-04-18 09:38 PM, Dave Hylands wrote:
>
> In cpp/src/command_classes/UserCode.cpp 
> <https://github.com/OpenZWave/open-zwave/pull/858#discussion_r60162622>:
>
> > @@ -264,7 +264,8 @@ bool UserCode::HandleMsg
> >  				memcpy( data, &_data[3], size );
> >  				value->OnValueRefreshed( data, size );
> >  			} else {
> > -				value->OnValueRefreshed( 0x00, 1);
> > +				size = 0;
> > +				value->OnValueRefreshed( (uint8 const*)&size, 1);
> >  			}
>
> I'm not familiar enough with the coding styles of the openzwave 
> maintainers, but personally, I would be inclined to code this using 
> something like the following (I try to avoid casts if I can).
>
> |uint8 zero = 0; value->OnValueRefreshed(&zero, 1); |
>
> but otherwise, your PR looks good. It might be worth mentioning that 
> the bug being fixed was introduced by commit 0d716a3 
> <https://github.com/OpenZWave/open-zwave/commit/0d716a3526eea2c23691c90073c3c5895af3bd24>
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub 
> <https://github.com/OpenZWave/open-zwave/pull/858/files/8dff23001459f71dada3898d78ac33574bdc608f#r60162622>
>